profiles/seccomp/kernel_linux.go

```diff
@@ -0,0 +1,96 @@
+package seccomp
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"golang.org/x/sys/unix"
+)
+
+// kernelVersion holds information about the kernel.
+type kernelVersion struct {
+	kernel uint // Version of the kernel (i.e., the "4" in "4.1.2-generic")
+	major  uint // Major revision of the kernel (i.e., the "1" in "4.1.2-generic")
+}
+
+func (k *kernelVersion) String() string {
+	return fmt.Sprintf("%d.%d", k.kernel, k.major)
+}
+
+var (
+	currentKernelVersion *kernelVersion
+	once                 sync.Once
+)
+
+// getKernelVersion gets the current kernel version.
+func getKernelVersion() (*kernelVersion, error) {
+	var err error
+	once.Do(func() {
+		var uts unix.Utsname
+		err = unix.Uname(&uts)
+		if err != nil {
+			return
+		}
+		// Remove the \x00 from the release for Atoi to parse correctly
+		currentKernelVersion, err = parseRelease(string(uts.Release[:bytes.IndexByte(uts.Release[:], 0)]))
+	})
+	return currentKernelVersion, err
+}
+
+// parseRelease parses a string and creates a kernelVersion based on it.
+func parseRelease(release string) (*kernelVersion, error) {
+	var (
+		version = kernelVersion{}
+		suffix  string
+	)
+
+	// Just make sure we get the "kernel" and "major revision". Sometimes we have
+	// 3.12.25-gentoo, but sometimes we just have 3.12-1-amd64. We add a bogus
+	// "-any-suffix" suffix to account for anything after "kernel.major" being
+	// optional, and may not be present in the input string.
+	parsed, err := fmt.Sscanf(release+"-any-suffix", "%d.%d%s", &version.kernel, &version.major, &suffix)
```

Comment:
this looks awfully hacky. why not just split by `.` and convert to int. That would be easy to read and same function can be used instead of the `parseMinVersion`

Reply:
You mean split by `.` + optionally split by `-`, because you also need to convert `3.12-1-amd64` to `3` and `12`.

Reply:
I'm fine with either that, or just scanf `%d.%d` and check `io.EOF` as explained in https://github.com/moby/moby/pull/41493/files#r497021178

Reply:
or regexp

Reply:
Let me give this another look; I added the `-any-suffix` hack because when I tried, it failed to perform the scan for some cases. However, I just tried changing back to simply `"%d.%d"`, and that worked. Trying to remember what case failed without the hack.